qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,709 @@
+
+    def _get_channel_ranges(self) -> List[str]:
+        """
+        A helper function that gets two channel names from the available
+        channels list and join them via a colon to define a channel range.
+        """
+        range_list = []
+        for i in self._get_slot_id():
+            channel = self.get_channels_by_slot(int(i))
+            for element in itertools.combinations(channel, 2):
+                range_list.append(':'.join(element))
+        return range_list
+
+    def get_channels(self) -> List[str]:
+        """
+        This function returns the name of the matrix channels.
+        User can call this function to see the names of the available
+        channels, in case he/she is not familiar with the naming convention.
+        However, note that, this is a standalone helper function and
+        the usage of channel attributes of the instrument driver does
+        not depend on the functionality of this method.
+        """
+        slot_id = self._get_slot_id()
+        row_list = self._get_rows()
+        column_list = self._get_columns()
+        matrix_channels = []
+        for i, slot in enumerate(slot_id):
+            for element in itertools.product(slot, row_list[i], column_list[i]):
+                matrix_channels.append(''.join(element))
+        return matrix_channels
+
+    def get_channels_by_slot(self, val: int) -> List[str]:
+        """
+        Returns the channel names of a given slot.
+
+        Args:
+            val: An integer value specifying the slot number.
+        """
+        slot_id = self._get_slot_id()
+        if str(val) not in slot_id:
+            raise UnknownOrEmptySlot("Please provide a valid slot identifier. "
+                                     f'Available slots are {slot_id}.')
+        row_list = self._get_rows()
+        column_list = self._get_columns()
+        matrix_channels_by_slot = []
+        for element in itertools.product(str(val), row_list[0], column_list[0]):
+            matrix_channels_by_slot.append(''.join(element))
+        return matrix_channels_by_slot
+
+    def get_analog_backplane_specifiers(self) -> List[str]:
+        """
+        Returns a list of comma separated strings representing available analog
+        backplane relays. This function should not be mixed with the
+        `get_backplane` method. The latter returns backplane relays which are
+        associated with a channel by using `set_backplane` method.
+        """
+        backplane_common_number = '9'
+        backplane_relay_common_numbers = ['11', '12', '13', '14', '15', '16']
+        slot_id = self._get_slot_id()
+        analog_backplane_relays = []
+        for element in itertools.product(slot_id, backplane_common_number,
+                                         backplane_relay_common_numbers):
+            analog_backplane_relays.append(''.join(element))
+        return analog_backplane_relays
+
+    def connect_row_to_columns(self, slot_id: int, row_id: int,
```

Comment:
i think since there are these `connect_*` convenience methods, it make sense to also have `disconnect_*` methods. At least in the drivers for other switch matrices in qcodes repo `connect` method also comes with a matching `disconnect`. Would you agree?

Reply:
Disconnect methods would be the replica of the connect methods with one line diff, thus bad code. I intentionally let these functions to return (which is not needed for it to function) the list of connected channels (i.e. opened channels) so that user pass these lists to close_channels methods directly without modification to disconnect. 

However, what can be done is that same functions can be used for both connect and disconnect via a boolean thus avoiding replication?